### PR TITLE
Added towers (not attacking yet, do disappear on hitting 0 health)

### PR DESCRIPTION
**Pull request by GwenTinho**

### CODE REVIEW

File: src/main/java/lol/server/CodeRunner.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package lol.server;
+
+public class CodeRunner {
```
[the remark this answers is not in the document]
Comment (ptal) [reply src/main/java/lol/server/CodeRunner.java:3]: Ubuntu + Sublime Text + Terminal FTW.

**Review by ptal — CHANGES_REQUESTED** [review]

File: src/main/java/lol/game/Team.java [hunk 2]
```diff
@@ -76,6 +89,12 @@ public void makeSpawnTurn(final Turn turn) {
         }
       }
     });
+    if(teamID == 0) {
+      turn.registerAction(new TowerSpawn(teamID, 3, 6));
```
[the remark this answers is not in the document]
Comment (AlexanderG97) [reply src/main/java/lol/game/Team.java:93]: To be honest I thought I might need to change this behavior in order to change the destroy behavior. Turned out not to be the case and this is equivalent.
I would say it's less hard-coded than before as now the position variables could be changed and could be set to some variable that changes with the size of the board.

File: src/main/java/lol/game/Team.java [hunk 3]
```diff
@@ -58,6 +68,9 @@ public boolean championAttack(int championID, int x, int y) {
     battlefield.visit(x, y, new TileVisitor(){
       @Override public void visitDestructible(Destructible d) {
         attacked[0] = champion.attack(d);
+        if(d.currentHP <= 0) {
```
[the remark this answers is not in the document]
Comment (ptal) [reply src/main/java/lol/game/Team.java]: it should stay protected I think. But the class `Team` cannot access protected member of a class it does not inherit from.

[the remark this answers is not in the document]
Comment (ptal) [reply src/main/java/lol/game/Team.java]: No, protected attributes are only accessible by the classes inheriting from the parent class. Where did you see that package access thing?

[the remark this answers is not in the document]
Comment (AlexanderG97) [reply src/main/java/lol/game/Team.java]: https://www.w3schools.com/java/ref_keyword_protected.asp#:~:text=The%20protected%20keyword%20is%20an,the%20same%20package%20and%20subclasses.
Should I change them back to protected?

[the remark this answers is not in the document]
Comment (ptal) [reply src/main/java/lol/game/Team.java]: Ok, I didn't know that, you just fixed a hole in my Java understanding, this rule doesn't apply to C++, I must have confused the two!
Normally, we don't access protected members across a package, as it weakens encapsulation.

You can keep it private, perhaps we won't need it protected after all.

[the remark this answers is not in the document]
Comment (AlexanderG97) [reply src/main/java/lol/game/Team.java]: OK, I guess it's ready to merge now?

File: src/main/java/lol/client/ai/RandomAITower.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package lol.client.ai;
+
+import java.util.*;
+import lol.game.*;
+import lol.game.action.*;
+
+public class RandomAITower extends AIBase {
```
Comment (ptal) [inline src/main/java/lol/client/ai/RandomAITower.java:7]: This class shares almost all its code with `RandomAI`. Please inherits from RandomAI, extract some methods (to attack nexus, and move) as protected methods, and use that in your `RandomAITower` to make it much smaller.

File: src/main/java/lol/game/Battlefield.java [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package lol.game;
 
+import javax.swing.text.html.Option;
```
Comment (ptal) [inline src/main/java/lol/game/Battlefield.java:3]: no need for this import?

File: src/main/java/lol/game/Tower.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package lol.game;
+
+public class Tower extends Attacker{
+    public static final int HP_TOWER = 5;
+    public static final int RANGE_OF_ATTACK = 2;
+    public static final int DAMAGE_TOWER = 4;
+    public static final int BLUE = 0;
```
Comment (ptal) [inline src/main/java/lol/game/Tower.java:7]: use Nexus.BLUE and Nexus.RED, this is to avoid mixing up the constant.

**Review by ptal — CHANGES_REQUESTED** [review]
We are almost there! Some very easy fix to be made and I merge.

**Pull request merged**